### ISSUE TITLE
--allow-hosts list is normalized but connection host is not

### ISSUE THREAD

**Issue by samdatkins** [miketheman/pytest-socket#246]
Hello,

I'm trying to block connections form everywhere except for localhost so that VS Code can still communicate with pytest. While working on this I've stumbled upon an issue where I noticed that if I use the arg `--allow-hosts=localhost`, it gets translated into `127.0.0.1` in the `normalize_allowed_hosts` function. This would be fine except the `host_from_connect_args` function does not normalize the host, which unfortunately means that code attempting to connect to "localhost" will always be blocked, even when attempting to allow it.

I've patched my local copy to work like this, and it's having the desired effect (note line 3):
```
def guarded_connect(inst, *args):
        host = host_from_connect_args(args)
        host = normalize_allowed_hosts([host])[0]
        if host in allowed_hosts or (
            _is_unix_socket(inst.family) and allow_unix_socket
        ):
...
```

Not sure if that's the ideal long-term solution but I wanted to present a working option. Thanks for maintaining this, let me know if you'd like me to create a PR with the change (or a similar one).

**Comment by jerr0328** [comment]
#275 is still awaiting review/merge, hoping this can get some attention 🤞

**Comment by andrelccorrea-blinctek** [comment]
Why is this PR still on hold?
Another vote in favor of the review and merge.

**Comment by andrelccorrea** [comment]
Hey guys, that's NOT me in the previous comment 😬.
Any updates?

**Comment by miketheman** [comment]
#275 merged and release in 0.7.0
